server/controllers/api/index.js

```diff
@@ -0,0 +1,18 @@
+import express from 'express';
+import login from './login_post';
+import addFlight from './addFlight_post';
+import flights from './flight_get';
+import updateFlight from './updateFlight_post';
+import editFlightInformation from './editFlightInformation_post';
+
+const router = express.Router();
+
+// reminder: routes naming should be edited later
+// on both clinet and server side!!
+router.post('/login_post', login);
+router.post('/addFlight_post', addFlight);
+router.get('/flight_get', flights);
+router.post('/updateflight_post', updateFlight);
+router.post('/editFlightInformation_post', editFlightInformation);
```

Comment:
If you have time, you could update these flight-related urls to follow REST patterns:
GET /<model> gets all of that type of model
POST /<model> creates a new type of that model
GET /<model>/:id gets one single model
POST (or sometimes PUT) /<model>/:id updates that instance

so here it would be something like
```
router.get('/flight', flights)
router.post('/flight', addFlight)
router.get('/flight/:flightId', getFlight) //I think you currently are using updateFlight more like a "getFlight"
router.post('/flight/:flightId, editFlightInformation)
```

For the last two, since `:flightId` is in the route, you'll be able to access the flight id within the request handler using `req.params.flightId`

I think the advantage here is that on the frontend, it's easy to figure out what URL you need to use to do something, without even having to open the backend code.  Some js frameworks, like backbone.js, almost force you into using this pattern (if you call model.save() on a backboneJs model instance, it automatically puts together that type of URL and posts).

Reply:
That's great, I was thinking about how to apply it the RESTful way.
I'll edit this when I get all the current work merged to the master.

About git-flow. Is it ok to make a PR with only the server part of a functionality or normally the PR should contain a  working feature( frontend + backend)?
I wonder how this works in professional companies?
@tomduggan85

Reply:
Cool.  Good question :) It depends on the size of the project / app.  In smaller projects it's often okay to make the PRs separate, because an engineer is probably still manually deploying code.  In medium or large projects, merging to master often triggers the deploy process automatically (usually to a staging environment, not directly to production) so more care has to be taken if the server change is a breaking change (like here, where the routes moved).

But then in large projects, the backend is often on different infrastructure / in a different repo entirely.

Reply:
Thanks, that clarifies things for me.